lib/beautiful/init.lua

```diff
@@ -186,22 +186,27 @@ end
 --   containing all the theme values.
 function beautiful.init(config)
     if config then
+        local state, t_config, t_theme = nil, nil, nil
         local homedir = os.getenv("HOME")
 
         -- If `config` is the path to a theme file, run this file,
         -- otherwise if it is a theme table, save it.
-        if type(config) == 'string' then
+        t_config = type(config)
+        if t_config == 'string' then
             -- Expand the '~' $HOME shortcut
             config = config:gsub("^~/", homedir .. "/")
             local dir = Gio.File.new_for_path(config):get_parent()
             rawset(beautiful, "theme_path", dir and (dir:get_path().."/") or nil)
             theme = protected_call(dofile, config)
-        elseif type(config) == 'table' then
+            t_theme = type(theme)
+            state = t_theme == 'table' and next(theme)
+        elseif t_config == 'table' then
             rawset(beautiful, "theme_path", nil)
             theme = config
+            state = next(theme)
```

Comment:
cc @Elv13 it seems test shim detecting this line as access to the theme so the tests are failing